pkg/cmd/agent-task/shared/log.go

```diff
@@ -0,0 +1,570 @@
+					return false, fmt.Errorf("failed to parse 'view' tool call arguments: %w", err)
+				}
+				fmt.Fprintf(w, "View %s\n", cs.Bold(relativeFilePath(args.Path)))
+
+				content := stripDiffFormat(choice.Delta.Content)
+
+				// TODO: Strip the diff formatting from this, but for now render as it is.
+				if err := renderFileContentAsMarkdown(args.Path, content, w, io); err != nil {
+					return false, fmt.Errorf("failed to render viewed file content: %w", err)
+				}
+			case "bash":
+				if v := unmarshal[bashToolArgs](args); v != nil {
+					if v.Description != "" {
+						renderToolCallTitle(w, cs, "Bash", v.Description)
+					} else {
+						renderToolCallTitle(w, cs, "Run Bash command", "")
+					}
+
+					contentWithCommand := choice.Delta.Content
+					if v.Command != "" {
+						contentWithCommand = fmt.Sprintf("%s\n%s", v.Command, choice.Delta.Content)
+					}
+					if err := renderFileContentAsMarkdown("commands.sh", contentWithCommand, w, io); err != nil {
+						return false, fmt.Errorf("failed to render bash command output: %w", err)
+					}
+				}
+			// TODO: consider including more details for these bash-related tool calls.
+			case "write_bash":
+				if v := unmarshal[writeBashToolArgs](args); v != nil {
+					renderToolCallTitle(w, cs, "Send input to Bash session", "")
+					continue
+				}
+			case "read_bash":
+				if v := unmarshal[readBashToolArgs](args); v != nil {
+					renderToolCallTitle(w, cs, "Read logs from Bash session", "")
+					continue
+				}
+			case "stop_bash":
+				if v := unmarshal[stopBashToolArgs](args); v != nil {
+					renderToolCallTitle(w, cs, "Stop Bash session", "")
+					continue
+				}
+			case "async_bash":
+				if v := unmarshal[asyncBashToolArgs](args); v != nil {
+					renderToolCallTitle(w, cs, "Start or send input to long-running Bash session", "")
+					continue
+				}
+			case "read_async_bash":
+				if v := unmarshal[readAsyncBashToolArgs](args); v != nil {
+					renderToolCallTitle(w, cs, "View logs from long-running Bash session", "")
+					continue
+				}
+			case "stop_async_bash":
+				if v := unmarshal[stopAsyncBashToolArgs](args); v != nil {
+					renderToolCallTitle(w, cs, "Stop long-running Bash session", "")
+					continue
+				}
+			case "think":
+				args := thinkToolArgs{}
+				if err := json.Unmarshal([]byte(tc.Function.Arguments), &args); err != nil {
+					return false, fmt.Errorf("failed to parse 'think' tool call arguments: %w", err)
+				}
+
+				// NOTE: omit the delta.content since it's the same as thought
+				renderToolCallTitle(w, cs, "Thought", "")
+				if err := renderRawMarkdown(args.Thought, w, io); err != nil {
+					return false, fmt.Errorf("failed to render thought: %w", err)
+				}
+			case "report_progress":
+				args := reportProgressToolArgs{}
+				if err := json.Unmarshal([]byte(tc.Function.Arguments), &args); err != nil {
+					return false, fmt.Errorf("failed to parse 'report_progress' tool call arguments: %w", err)
+				}
```

Comment:
Basically the same concern. Given the high rate of changes happening everyday, I think we should just be ready to print a generic record when we can't parse the JSON args. There are also other cases down here.